### PR TITLE
Update python testing in linux-ci

### PR DESCRIPTION
**Pull request by vxgmichel**

### CODE REVIEW

File: .github/workflows/linux-ci.yml [hunk 1]
```diff
@@ -35,10 +35,25 @@ jobs:
 
   python:
     runs-on: ubuntu-22.04
+    env:
+      PYTEST_BASE_ARGS: |
```
Comment (FirelightFlagboy) [inline .github/workflows/linux-ci.yml:39]:
```suggestion
      PYTEST_BASE_ARGS: >-
```

File: .github/workflows/linux-ci.yml [hunk 2]
```diff
@@ -57,19 +72,37 @@ jobs:
...
-          echo "::warning title=Libfuse path::use lib found at $LIBFUSE_PATH"
-          echo "FUSE_LIBRARY_PATH=$LIBFUSE_PATH" >> "$GITHUB_ENV"
+        run: poetry install -E core -E backend --remove-untracked
 
       - name: Generate PyQT GUI data
         run: poetry run python misc/generate_pyqt.py
 
       - name: Install libparsec_python
+        if: matrix.binding == 'rust-binding'
         run: poetry run pip install oxidation/libparsec_python
```
Comment (FirelightFlagboy) [inline .github/workflows/linux-ci.yml:82]: `pip install` will generate a release build, we may want to use a debug build which is faster to build

File: .github/workflows/linux-ci.yml [hunk 3]
```diff
@@ -57,19 +72,37 @@ jobs:
...
+        run: timeout 200 poetry run pytest $PYTEST_BASE_ARGS tests --runrust -m rust
+
+      - name: Mountpoint tests
+        run: timeout 200 poetry run pytest $PYTEST_BASE_ARGS tests --runmountpoint --runslow -m mountpoint
+
+      - name: GUI tests
+        run: |
+          # Required dependencies for Qt testing. See:
+          # https://pytest-qt.readthedocs.io/en/latest/troubleshooting.html#github-actions
+          sudo apt-get install libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0 x11-utils
```
Comment (FirelightFlagboy) [inline .github/workflows/linux-ci.yml:98]: from the link, you're missing the line

```bash
/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1920x1200x24 -ac +extension GLX
```

It's expected ?

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]
for steps using `timeout` [job timeout in Github Action](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes)

File: .github/workflows/linux-ci.yml (hunk 2 above)
Comment (vxgmichel) [reply .github/workflows/linux-ci.yml:82]: but then it should be faster to test right?

File: .github/workflows/linux-ci.yml (hunk 3 above)
Comment (vxgmichel) [reply .github/workflows/linux-ci.yml:98]: > you're missing the line [...]

This is for Travis CI as far as I understand.

Comment (vxgmichel) [reply .github/workflows/linux-ci.yml:98]: I'm counting on `pytest-xvfb` to deal with this kind of stuff, and it seems to work so far

File: .github/workflows/linux-ci.yml (hunk 1 above)
Comment (vxgmichel) [reply .github/workflows/linux-ci.yml]: [3370cad](https://github.com/Scille/parsec-cloud/pull/2652/commits/3370cad5373b0ca815121d0c7ec30fdc50047de1)

**Review by FirelightFlagboy — COMMENTED** [review]
We don't want to disable the Job on `Azure-Pipeline`, we will decommission it later

**Comment by vxgmichel** [comment]
> We don't want to disable the Job on `Azure-Pipeline`, we will decommission it later

That's why the commit is called `[Disable azure while testing](https://github.com/Scille/parsec-cloud/pull/2652/commits/47819757c8cd4f7f77027e586c614c87f5e6e304)` ;)

File: .github/workflows/linux-ci.yml [hunk 4]
```diff
@@ -50,11 +52,10 @@ jobs:
...
 
       - name: Install APT packages
-        run: |
-          # Only libfuse3 is available on ubuntu-22.04,
-          # and `desktop-file-utils` provides `update-desktop-database` used by `tests/scripts/run_testenv.sh`
-          # and a specific version of postgresql is used
-          sudo apt-get -y install libfuse2 fuse desktop-file-utils postgresql-${{ env.PGVERSION }}
+        # Only libfuse3 is available on ubuntu-20.04,
+        # and `desktop-file-utils` provides `update-desktop-database` used by `tests/scripts/run_testenv.sh`
+        # and a specific version of postgresql is used
```
Comment (FirelightFlagboy) [inline .github/workflows/linux-ci.yml:57]: I would prefer the comment above `- name: Install ...`

Comment (vxgmichel) [reply .github/workflows/linux-ci.yml:57]: I guess it's a matter of opinion :shrug:

Comment (vxgmichel) [reply .github/workflows/linux-ci.yml:57]: I guess it's a matter of opinion :shrug:

File: .github/actions/setup-rust/action.yml [hunk 5]
```diff
@@ -41,12 +41,18 @@ outputs:
...
+      run: echo "::set-output name=image::$ImageOS"
+      shell: bash
     - name: Cache Rust
       id: cache-rust
       uses: actions/cache@c3f1317a9e7b1ef106c153ac8c0f00fed3ddbc0d # pin v3.0.4
       with:
-        key: ${{ runner.os }}-rust-${{ inputs.version }}-${{ inputs.cache-key }}-${{ hashFiles('**/Cargo.lock') }}
-        restore-keys: ${{ runner.os }}-rust-${{ inputs.version }}-${{ inputs.cache-key }}
+        key: ${{ runner.os }}-${{ steps.os.outputs.image }}-rust-${{ inputs.version }}-${{ inputs.cache-key }}-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: ${{ runner.os }}-${{ steps.os.outputs.image }}-rust-${{ inputs.version }}-${{ inputs.cache-key }}
```
Comment (FirelightFlagboy) [inline .github/actions/setup-rust/action.yml:55]:
```suggestion
        key: ${{ runner.os }}-${{ env.ImageOS }}-rust-${{ inputs.version }}-${{ inputs.cache-key }}-${{ hashFiles('**/Cargo.lock') }}
        restore-keys: ${{ runner.os }}-${{ env.ImageOS }}-rust-${{ inputs.version }}-${{ inputs.cache-key }}
```

File: .github/actions/setup-rust/action.yml [hunk 6]
```diff
@@ -41,12 +41,18 @@ outputs:
 runs:
   using: composite
   steps:
+    # Set steps.os.outputs.image to the specific OS (e.g. 'ubuntu20')
+    # see https://github.com/actions/cache/issues/543
+    - name: Get operating system name and version.
+      id: os
+      run: echo "::set-output name=image::$ImageOS"
+      shell: bash
```
Comment (FirelightFlagboy) [inline .github/actions/setup-rust/action.yml:49]:
```suggestion
```

You can directly use `env.<variable_name>` see comment below

File: tests/core/gui/test_workspace_button.py [hunk 7]
```diff
@@ -273,6 +273,8 @@ async def test_workspace_button_delete_clicked(qtbot, workspace_fs, core_config,
 @pytest.mark.gui
 @pytest.mark.trio
 async def test_workspace_button_timestamped(qtbot, workspace_fs, core_config, alice_user_info):
+    if IS_OXIDIZED:
+        pytest.xfail("TODO: implement `.to_timestamped` method in rust")
```
Comment (FirelightFlagboy) [inline tests/core/gui/test_workspace_button.py:277]:
```suggestion
@pytest.mark.xfail(IS_OXIDIZED, reason="TODO: implement `.to_timestamped` method in rust")
async def test_workspace_button_timestamped(qtbot, workspace_fs, core_config, alice_user_info):
```
see [conditional parameter](docs.pytest.org/en/7.1.x/how-to/skipping.html#condition-parameter)

File: tests/core/gui/test_validators.py [hunk 8]
```diff
@@ -284,9 +285,11 @@ def test_file_name_validator(qtbot, core_config):
 )
 def test_user_name_validator(qtbot, core_config, name_and_expected):
     switch_language(core_config, "en")
-
     user_name, expected = name_and_expected
 
+    if IS_OXIDIZED and not expected:
+        pytest.xfail("TODO: Add validation for user name in rust")
+
```
Comment (FirelightFlagboy) [inline tests/core/gui/test_validators.py:292]: use [conditional parameter for `xfail`](https://docs.pytest.org/en/7.1.x/how-to/skipping.html#condition-parameter)

File: tests/core/mountpoint/test_folder_operations.py [hunk 9]
```diff
@@ -90,9 +91,11 @@ def __truediv__(self, path):
 @pytest.mark.mountpoint
 @pytest.mark.flaky(reruns=0)
 def test_folder_operations(tmpdir, caplog, hypothesis_settings, mountpoint_service_factory):
-
     tentative = 0
 
+    if IS_OXIDIZED:
+        pytest.xfail("TODO: investigate `database is locked` error")
+
```
Comment (FirelightFlagboy) [inline tests/core/mountpoint/test_folder_operations.py:98]: use [conditional parameter for `xfail`](https://docs.pytest.org/en/7.1.x/how-to/skipping.html#condition-parameter)

File: tests/fuzzing/test_fuzz_core.py [hunk 10]
```diff
@@ -233,6 +234,8 @@ async def _fuzzer_cmd(id, core, workspace, fs_state):
 @pytest.mark.trio
 @pytest.mark.slow
 async def test_fuzz_core(request, running_backend, alice_core):
+    if IS_OXIDIZED:
+        pytest.xfail("TODO: fix database is locked error in rust")
```
Comment (FirelightFlagboy) [inline tests/fuzzing/test_fuzz_core.py:238]:
```suggestion
@pytest.mark.slow
@pytest.mark.xfail(IS_OXIDIZED, reason="TODO: fix database is locked error in rust")
async def test_fuzz_core(request, running_backend, alice_core):
```

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: .github/actions/setup-rust/action.yml (hunk 5 above)
Comment (vxgmichel) [reply .github/actions/setup-rust/action.yml]: 7f7f1482873b86098b805c9b499fce615d613520

File: .github/actions/setup-rust/action.yml (hunk 6 above)
Comment (vxgmichel) [reply .github/actions/setup-rust/action.yml]: 7f7f1482873b86098b805c9b499fce615d613520

File: tests/core/gui/test_workspace_button.py (hunk 7 above)
Comment (vxgmichel) [reply tests/core/gui/test_workspace_button.py]: e222d41bd71b49becac04b7170b75e4a8ac4a92b

File: tests/core/gui/test_validators.py (hunk 8 above)
Comment (vxgmichel) [reply tests/core/gui/test_validators.py]: e222d41bd71b49becac04b7170b75e4a8ac4a92b

File: tests/fuzzing/test_fuzz_core.py (hunk 10 above)
Comment (vxgmichel) [reply tests/fuzzing/test_fuzz_core.py]: e222d41bd71b49becac04b7170b75e4a8ac4a92b

File: tests/core/mountpoint/test_folder_operations.py [hunk 11]
```diff
@@ -89,12 +89,13 @@ def __truediv__(self, path):
 
 @pytest.mark.slow
 @pytest.mark.mountpoint
+@pytest.mark.xfail(IS_OXIDIZED, reason="TODO: investigate `database is locked` error")
 @pytest.mark.flaky(reruns=0)
 def test_folder_operations(tmpdir, caplog, hypothesis_settings, mountpoint_service_factory):
     tentative = 0
 
     if IS_OXIDIZED:
```
Comment (FirelightFlagboy) [inline tests/core/mountpoint/test_folder_operations.py:97]:
```suggestion
```

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Comment by FirelightFlagboy** [comment]
this issue closes #2648

**Pull request merged**